devnet-sdk/testing/testlib/validators/forks.go

```diff
@@ -0,0 +1,137 @@
+package validators
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/devnet-sdk/system"
+	"github.com/ethereum-optimism/optimism/devnet-sdk/testing/systest"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+)
+
+// getForkConfig is a helper function that retrieves the ForkConfig for a specific L2 chain.
+func getForkConfig(t systest.T, sys system.System, chainIdx uint64) (system.ChainConfig, *uint64, error) {
+	if len(sys.L2s()) <= int(chainIdx) {
+		return nil, nil, fmt.Errorf("chain index %d out of range, only %d L2 chains available", chainIdx, len(sys.L2s()))
+	}
+
+	chain := sys.L2s()[chainIdx]
+
+	chainConfig, err := chain.Config()
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get chain config for L2 chain %d: %w", chainIdx, err)
+	}
+
+	block, err := chain.BlockByNumber(t.Context(), nil)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get latest block for L2 chain %d: %w", chainIdx, err)
+	}
+
+	timestamp := block.Time()
+
+	return chainConfig, &timestamp, nil
+}
+
+// forkConfigValidator is a helper function that checks if a specific L2 chain meets a fork condition.
+func forkConfigValidator(chainIdx uint64, forkName rollup.ForkName, shouldBeActive bool, forkConfigMarker interface{}) systest.PreconditionValidator {
+	return func(t systest.T, sys system.System) (context.Context, error) {
+		forkConfig, timestamp, err := getForkConfig(t, sys, chainIdx)
+		if err != nil {
+			return nil, err
+		}
+
+		isActive := forkConfig.IsForkActivated(forkName, *timestamp)
+		if isActive != shouldBeActive {
+			if shouldBeActive {
+				return nil, fmt.Errorf("L2 chain %d does not have fork %s activated, which it should be for this validator to pass", chainIdx, forkName)
+			} else {
+				return nil, fmt.Errorf("L2 chain %d has fork %s activated, but it should not be for this validator to pass", chainIdx, forkName)
+			}
+		}
+
+		return context.WithValue(t.Context(), forkConfigMarker, forkConfig), nil
+	}
+}
+
+// ChainConfigGetter is a function type that retrieves a ForkConfig from a context.
+type ChainConfigGetter = func(context.Context) system.ChainConfig
+
+// AcquireForkConfig returns a ForkConfigGetter and a PreconditionValidator
+// that ensures a ForkConfig is available for the specified L2 chain.
+// The ForkConfig can be used to check if various forks are activated.
+func acquireForkConfig(chainIdx uint64, forkName rollup.ForkName, shouldBeActive bool) (ChainConfigGetter, systest.PreconditionValidator) {
+	chainConfigMarker := new(byte)
+	validator := forkConfigValidator(chainIdx, forkName, shouldBeActive, chainConfigMarker)
+	return func(ctx context.Context) system.ChainConfig {
+		return ctx.Value(chainConfigMarker).(system.ChainConfig)
+	}, validator
+}
+
+// RequiresFork returns a validator that ensures a specific L2 chain has a specific fork activated.
+func AcquireRequiresL2Fork(chainIdx uint64, forkName rollup.ForkName) (ChainConfigGetter, systest.PreconditionValidator) {
```

Comment:
nit: the AcquireRequires* helpers  read funny

maybe AcquireL2WithFork and so on?